pyvisa/util.py

```diff
@@ -930,49 +968,15 @@ def get_debug_info(to_screen: Literal[False]) -> str:
     pass
 
 
-def get_debug_info(to_screen=True):
+def get_debug_info(to_screen: bool = True):
     """Get the PyVISA debug information."""
     out = system_details_to_str(get_system_details())
     if not to_screen:
         return out
     print(out)
 
 
-machine_types = {
-    0: "UNKNOWN",
-    0x014C: "I386",
-    0x0162: "R3000",
-    0x0166: "R4000",
-    0x0168: "R10000",
-    0x0169: "WCEMIPSV2",
-    0x0184: "ALPHA",
-    0x01A2: "SH3",
-    0x01A3: "SH3DSP",
-    0x01A4: "SH3E",
-    0x01A6: "SH4",
-    0x01A8: "SH5",
-    0x01C0: "ARM",
-    0x01C2: "THUMB",
-    0x01C4: "ARMNT",
-    0x01D3: "AM33",
-    0x01F0: "POWERPC",
-    0x01F1: "POWERPCFP",
-    0x0200: "IA64",
-    0x0266: "MIPS16",
-    0x0284: "ALPHA64",
-    # 0x0284: 'AXP64', # same
-    0x0366: "MIPSFPU",
-    0x0466: "MIPSFPU16",
-    0x0520: "TRICORE",
-    0x0CEF: "CEF",
-    0x0EBC: "EBC",
-    0x8664: "AMD64",
-    0x9041: "M32R",
-    0xC0EE: "CEE",
-}
-
-
-def get_shared_library_arch(filename: str) -> str:
+def get_shared_library_arch(filename: Path) -> PEMachineType:
```

Comment:
The type annotation is overly restrictive here
```suggestion
def get_shared_library_arch(filename: str | Path) -> PEMachineType:
```
or get the annotation from io